### PR TITLE
Added the S3 bucket finder in passive scripts

### PR DESCRIPTION
**Pull request by alishasinghania**

### CODE REVIEW

**Comment by kingthorin** [comment]
To address the DCO requirement you'll need to sign-off the commit(s):

- https://github.com/zaproxy/zaproxy/blob/develop/CONTRIBUTING.md#developer-certificate-of-origin
- https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

**Comment by thc202** [comment]
The changelog should be updated.

File: passive/s3.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// S3 bucket finder by alishasinghania09@gmail.com 
+
+function scan(ps, msg, src) {
+    // populate some parameters which will be needed if s3 bucket url is present
+    var alertRisk = 1
+    var alertReliability = 4
+    var alertTitle = 's3 bucket url'
```
Comment (kingthorin) [inline passive/s3.js:7]:
```suggestion
    var alertTitle = 's3 Bucket URL'
```

File: passive/s3.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    var alertDesc = 's3 bucket url found in response'
```
Comment (kingthorin) [inline passive/s3.js:8]:
```suggestion
    var alertDesc = 's3 bucket URL found in response.'
```

File: passive/s3.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    var alertSolution = 'Remove s3 buckets name from response or make sure the permissions in bucket are configured properly'
```
Comment (kingthorin) [inline passive/s3.js:9]:
```suggestion
    var alertSolution = 'Remove s3 buckets name from response or make sure the permissions in bucket are configured properly.'
```

File: passive/s3.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    var cweId = 0
+    var wascId = 0
```
Comment (kingthorin) [inline passive/s3.js:11]:
```suggestion
    var cweId = 200
    var wascId = 13
```

Information leakage: http://projects.webappsec.org/w/page/13246975/Threat%20Classification%20Taxonomy%20Cross%20Reference%20View

File: passive/s3.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    else{
+        // test the regex against the message body
+        var body = msg.getResponseBody().toString()
+        if (re.test(body)) {
+            re.lastIndex = 0 
+            var founds3bucket = []
+            var comm
+            while (comm = re.exec(body)) {
+                founds3bucket.push(comm[0]);
+            }
```
Comment (kingthorin) [inline passive/s3.js:34]:
```suggestion
            var bucket
            while (bucket = re.exec(body)) {
                founds3bucket.push(bucket[0]);
            }
```

No longer dealing with comments.

**Review by kingthorin — CHANGES_REQUESTED** [review]

File: passive/s3.js (hunk 5 above)
Comment (alishasinghania) [reply passive/s3.js]: Made the suggested changes in https://github.com/zaproxy/community-scripts/pull/213/commits/8b9fcfbef900d1c852e6c98de6fffce7ec5bf6eb commit.

File: passive/s3.js (hunk 4 above)
Comment (alishasinghania) [reply passive/s3.js]: Made the suggested changes in https://github.com/zaproxy/community-scripts/pull/213/commits/8b9fcfbef900d1c852e6c98de6fffce7ec5bf6eb commit

File: passive/s3.js (hunk 3 above)
Comment (alishasinghania) [reply passive/s3.js]: Made the suggested changes in https://github.com/zaproxy/community-scripts/pull/213/commits/8b9fcfbef900d1c852e6c98de6fffce7ec5bf6eb commit

File: passive/s3.js (hunk 2 above)
Comment (alishasinghania) [reply passive/s3.js]: Made the suggested changes in https://github.com/zaproxy/community-scripts/pull/213/commits/8b9fcfbef900d1c852e6c98de6fffce7ec5bf6eb commiy.

File: passive/s3.js (hunk 1 above)
Comment (alishasinghania) [reply passive/s3.js]: Made the suggested changes in https://github.com/zaproxy/community-scripts/pull/213/commits/8b9fcfbef900d1c852e6c98de6fffce7ec5bf6eb commit.

File: passive/s3.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+// S3 bucket finder by alishasinghania09@gmail.com 
+
+function scan(ps, msg, src) {
+    // populate some parameters which will be needed if s3 bucket url is present
+    var alertRisk = 1
+    var alertReliability = 4
+    var alertTitle = 's3 Bucket URL'
+    var alertDesc = 's3 Bucket URL found in response'
```
Comment (kingthorin) [inline passive/s3.js:8]:
```suggestion
    var alertDesc = 's3 Bucket URL found in response.'
```

**Comment by kingthorin** [comment]
Thanks, please also update the https://github.com/zaproxy/community-scripts/blob/master/CHANGELOG.md like @thc202 mentioned.

In the topmost `### Added` section, something like `- Passive scan script for finding potential s3 bucket URLs.

**Review by kingthorin — APPROVED** [review]
Thanks

**Comment by alishasinghania** [comment]
> The changelog should be updated.

@thc202 changelog has been updated. Please approve the pull request.

File: passive/s3.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+// S3 bucket finder by alishasinghania09@gmail.com 
+
+function scan(ps, msg, src) {
+    // populate some parameters which will be needed if s3 bucket url is present
+    var alertRisk = 1
+    var alertReliability = 4
```
Comment (thc202) [inline passive/s3.js:6]: - This should be 3 at most, 4 is for the user to set. The scan rule can't confirm that the URL is an actual leakage, e.g. docs, examples.
 - `Reliability` → `Confidence`

**Comment by thc202** [comment]
Commits can be squashed.

**Comment by alishasinghania** [comment]
@thc202 I have made the changes and mistakenly closed this pull request. I want to reopen this pull request but not able to do so.

**Comment by kingthorin** [comment]
I believe it’s because you deleted the branch the PR was based on. I know you have since restored it but it seems that isn’t enough for GitHub.

**Comment by thc202** [comment]
@alishasinghania feel free to raise a new PR if you can't reopen this one.

**Pull request closed without merging**